runtime.md

```diff
@@ -23,37 +14,127 @@ This allows the hooks to perform cleanup and teardown logic after the runtime de
+1. OCI compliant runtime is invoked with a reference to the location of the bundle.
+   How this reference is passed to the runtime is an implementation detail.
+2. The container's runtime environment MUST be created according to the configuration in `config.json` and `runtime.json`.
+   Any updates to `config.json` or `runtime.json` after container is running MUST not affect the container.
+3. The prestart hooks MUST be invoked by the runtime.
+   If any prestart hook fails, then the container MUST be stopped and the lifecycle continues at step 8.
+4. The user specified process MUST be executed in the container.
+5. The poststart hooks MUST be invoked by the runtime.
+   If any poststart hook fails, then the container MUST be stopped and the lifecycle continues at step 8.
+6. Additional actions such as pausing the container, resuming the container or signaling the container MAY be performed using the runtime interface.
+   The container MAY also error out, exit or crash.
+7. The container MUST be destroyed by undoing the steps performed during create phase (step 2).
+8. The poststop hooks MUST be invoked by the runtime and errors, if any, MAY be logged.
 
 Note: The lifecycle is a WIP and it will evolve as we have more use cases and more information on the viability of a separate create phase.
 
+## Operations
+
+OCI compliant runtimes MUST support the following operations, unless the operation is not supported by the base operating system.
+
+### Errors
+In cases where the specified operation generates an error, this specification does not mandate how, or even if, that error is returned or exposed to the user of an implementation.
+Unless otherwise stated, generating an error MUST leave the state of the environment as if the operation were never attempted - modulo any possible trivial ancillary changes such as logging.
+
+### Query State
+
+`state <container-id>`
+
+This operation MUST generate an error if it is not provided the ID of a container.
+This operation MUST return the state of a container as specified in the [State](#state) section.
+
+
+### Start
+
+`start <path-to-bundle> <container-id>`
+
+This operation MUST generate an error if it is not provided a path to the bundle and the container ID to associate with the container.
+If the ID provided is not unique across all containers running on this host, or is not valid in any other way, the implementation MUST generate an error.
+Using the data in `config.json` and `runtime.json` files, that are in the bundle's directory, this operation MUST create a new container.
+This includes creating the relevant namespaces, resource limits, etc and configuring the appropriate capabilities for the container.
+A new process within the scope of the container  MUST be created as specified by the `config.json` file.
+The current working directory of the process MUST be set to the `cwd` field of the config.
+If the path specified by the `cwd` fields does not exist then an error MUST be generated.
```

Comment:
On Thu, Jan 21, 2016 at 10:35:22AM -0800, Doug Davis wrote:
> > +A new process within the scope of the container  MUST be created as specified by the `config.json` file.
> > +The current working directory of the process MUST be set to the `cwd` field of the config.
> > +If the path specified by the `cwd` fields does not exist then an error MUST be generated.
> 
> I think its important to say it here because this is where people
> might think its ok to create the dir for them.

Hopefully runtime-authors implementing ‘cwd’ read both this section
and the ‘cwd’ section [1] ;).  And if we don't mention ‘cwd’ here, the
only place bundle-authors will hear about it is [1].  So I'd rather
leave it out here and avoid splitting the normative ‘cwd’ spec between
two sections.

[1]: https://github.com/opencontainers/specs/blob/v0.2.0/config.md#process-configuration